src/algebra/gcd_monoid/integrally_closed.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebra.gcd_monoid.basic
+import ring_theory.integrally_closed
+import ring_theory.polynomial.scale_roots
+import ring_theory.polynomial.eisenstein
+
+/-!
+
+# GCD domains are integrally closed
+
+-/
+
+open_locale big_operators polynomial
+
+lemma is_localization.surj_of_gcd_domain {R : Type*} [comm_ring R] [is_domain R] [gcd_monoid R]
+  (M : submonoid R) {K} [comm_semiring K] [algebra R K] [is_localization M K] (z : K) :
```

Comment:
```suggestion
  (M : submonoid R) {K : Type*} [comm_ring K] [algebra R K] [is_localization M K] (z : K) :
```
`K` is automatically a ring, since it is an `R`-algebra, it is confusing to only assume it to be a `semiring`.